backend/api/parse_pdf.py

```
@@ -338,19 +342,366 @@ async def parse_with_llamaparse(file: UploadFile, session_id: str = None) -> str
+            ScenarioPersona.deleted_at.is_(None)
+        ).all()
+        has_personas_with_images = any(persona.image_url for persona in all_personas_final)
+        
+        # Set images_completed if either scenes or personas have images
+        scenario.images_completed = has_scenes_with_images or has_personas_with_images
+        debug_log(f"[FULL_SAVE] Images check - Scenes with images: {has_scenes_with_images}, Personas with images: {has_personas_with_images}, images_completed: {scenario.images_completed}")
+        
+        db.commit()
+        debug_log(f"[FULL_SAVE] Successfully saved full data for scenario {scenario_id}")
+        
+    except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)